src/components/ItemsTable.tsx

```diff
@@ -272,22 +278,28 @@ export const ItemsTable = <
 		...restProps
 	} = props;
 
+	const [isDownloading, setDownloading] = useState(false);
+
 	return (
 		<div {...restProps} data-class="table">
 			{getExportCSV && (
 				<div css={csvDownload}>
+					{isDownloading && <img src={LoadingSpinner} css={spinnerStyle} />}
 					<Button
 						size="small"
 						variant="outlined"
 						color="secondary"
+						disabled={isDownloading}
 						onClick={async () => {
+							setDownloading(true);
 							const { columns, data, filename } = await getExportCSV();
 							const csvConfig = mkConfig({
 								columnHeaders: columns,
 								filename,
 							});
 							const csv = generateCsv(csvConfig)(data);
 							download(csvConfig)(csv);
```

Comment:
```suggestion
							await download(csvConfig)(csv);
```
don't we need `await` here?

Reply:
`export declare const download: (config: ConfigOptions) => (csvOutput: CsvOutput) => IO;`

download function is not an async function.